ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -582,6 +590,49 @@ object Queries {
     (allTpids diff grouped.keySet).view.map((_, Map.empty[Party, Off])).toMap ++ grouped
   }
 
+  // invariant: each element x of result has `x.values.flatten.size <= size`
+  private[dbbackend] def chunkBySetSize[K, V](
+      size: Int,
+      groups: NonEmpty[Map[K, NonEmpty[Set[V]]]],
+  ): Vector[NonEmpty[Map[K, NonEmpty[Set[V]]]]] = {
```

Comment:
would it not be easier to allow for chunks to be empty and, if empty, not delete?

Reply:
Unfortunately, we would implement that by filtering them out, because the SQL expression won't work on empty things.

Reply:
(I meant not to have a SQL expression in the case the chunk is empty, but it's not a big deal)